R/family.R

```diff
@@ -0,0 +1,88 @@
+#' Define `epidist` family
+#'
+#' This function is used within [epidist()] to create a model specific custom
+#' `brms` family object. This object is passed to `brms`. It is unlikely that
+#' as a user you will need this function, but we export it nonetheless to be
+#' transparent about what exactly is happening inside of a call to [epidist()].
+#'
+#' @param data A `data.frame` containing line list data
+#' @param family Output of a call to `brms::brmsfamily()`
+#' @param ... ...
+#'
+#' @family family
+#' @export
```

Comment:
do we actually need to export this in order to be transparent? I think its a judgement call but I would be okay with no doing so as long as it has docs?

Reply:
Don't have strong views about the exporting of the internal methods here. I think we are currently exporting most of these internal things. This is like a 2-stage internal. `epidist_family` is 1-stage internal then `epidist_family_model` is 2-stage and `epidist_family_reparam` is 2-stage. Just export 1-stage? Or export all stage? Or no export? Don't really mind as long as consistent. Feel fine with exporting all for now

Reply:
So i thought the idea was that we were exporting everything a user might need to change if adding their own family or model? As this is no longer s3 (and hence not something we expect them to extend) that prompted me to suggest we don't export it

Reply:
I think we need to export epidist_family_model because we expect the user to maybe have to write their own extensions of it. I don't feel that strongly but have been pushed over time to always try and not export things if possible

Reply:
Right got you, understand where you are coming from in terms of what are the operations we might expect users to contribute to. Maybe it seems hard for them to develop their new model `epidist_family_model` without having these exported? I don't know, they could just use `:::`

Reply:
yes exactly hence thinking we need `epidist_family_model` to be exported but maybe not `epidist_family`. I don't feel strongly either way

Reply:
I think this is a broader issue and there are other cases outside of this one which a decision would apply to, so I'd like to put this into a new issue

Reply:
https://github.com/epinowcast/epidist/issues/79